march_hardware/include/march_hardware/imotioncube/motor_controller.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Project March.
+
+#include "actuation_mode.h"
+#include "motor_controller_state.h"
+#include <string>
+
+namespace march
+{
+class MotorController
+{
+public:
+  virtual double getAngleRadAbsolute() = 0;
+  virtual double getAngleRadIncremental() = 0;
+  virtual double getVelocityRadAbsolute() = 0;
+  virtual double getVelocityRadIncremental() = 0;
```

Comment:
What if my motor controller does not have either an absolute or incremental encoder?

Reply:
I thought about this; the incremental/absolute encoders are a characteristic of Joint rather than of the motor controller, so it would make more sense to move such specifications there. That would require me to subclass Joint into different types, which I preferred to avoid, especially since our current three options for motor controllers will probably be connected to both types of encoders. I think such a refactor of the joint class is better postponed to a moment in the future where an encoder is actually connected to something other than a motor controller.

Reply:
Hmm on second thought, I could move the calculations using both encoders downstream to the motor controller. Then these would simplify to getPosition() and getVelocity(). Downside would be that these calculations would be that this code would probably be duplicate in each different MotorController class, but that might again be resolvable by inheriting from a MultipleEncodersMotorControllers class that implements these methods common for motor controllers with both encoders.

That was a nice spam of hersenspinsels, let me know if it makes any sense and what you think.

Reply:
Decided that since the 3 motor controllers currently under consideration all have both an incremental and an absolute encoder, this change is not very useful for now. It won't be difficult to implement later on when such a motor controller is actually put to use.